core/services/feeds/service.go

```diff
@@ -331,3 +395,36 @@ func (s *service) getCSAPrivateKey() (privkey []byte, err error) {
 func (s *service) Unsafe_SetFMSClient(client pb.FeedsManagerClient) {
 	s.fmsClient = client
 }
+
+type GenericJobSpec struct {
+	Type          job.Type    `toml:"type"`
+	SchemaVersion uint32      `toml:"schemaVersion"`
+	Name          null.String `toml:"name"`
+}
+
+func (s *service) generateJob(spec string) (*job.Job, error) {
+	genericJS := GenericJobSpec{}
+	err := toml.Unmarshal([]byte(spec), &genericJS)
+	if err != nil {
+		return nil, ErrTOMLParseError
```

Comment:
hm might be still good to log or otherwise pass the parsing error somehow? unless we don't need any more detailed info

Reply:
Changed this to be a wrapped error instead